packages/gatsby/src/utils/webpack.config.js

```diff
@@ -677,7 +677,15 @@ module.exports = async (
         },
         // If a chunk is used in at least 2 components we create a separate chunk
         shared: {
-          test: module => !isCssModule(module),
+          test(module, { chunkGraph }) {
+            for (const chunk of chunkGraph.getModuleChunksIterable(module)) {
+              if (chunk.canBeInitial()) {
+                return false
+              }
+            }
+
+            return !isCssModule(module)
+          },
```

Comment:
this might need to be either reverted or reworked. A lot of our split chunking rules makes very little sense with client components approach - especially ones (not this particular one) that check for usage in "all" or at least 2 page templates (as page templates might not be there anymore etc).